### PR TITLE
ST46607 - Dialog fixed, Boss edit

### PR DESCRIPTION
**Pull request by jame581**
1. Employee dialog fixed 

- Change form board style to FixedDialog

- This dialog is used for editing existing boss  

2. Boss edit

- Button added, now you can edit boss

3. Listbox for employees added

- Prepared for implementation of show employees of selected boss

 
More info in commit.

### CODE REVIEW

File: INPTP_AppForFixing/EmployeeDialog.cs [hunk 1]
```diff
@@ -1,59 +1,78 @@
...
+                dateTimePickerBirthDate.Value = boss.OurBirthDate;
             }
         }
 
         private void btnOk_Click(object sender, EventArgs e)
         {
             if (testOfValues())
             {
-                main.Bosses.Add(new Boss(Int32.Parse(tBID.Text), tBFirstName.Text, tBLastName.Text, tBJob.Text, dateTimePickerBirthDate.Value, Double.Parse(tBSalary.Text), new Department(tBDpName.Text)));
+                int Id = Int32.Parse(tBID.Text);
```
Comment (rodi0878) [inline INPTP_AppForFixing/EmployeeDialog.cs]: id not Id

File: INPTP_AppForFixing/EmployeeDialog.cs [hunk 2]
```diff
@@ -1,59 +1,78 @@
...
+                Boss newB = new Boss(Id, tBFirstName.Text, tBLastName.Text, tBJob.Text, dateTimePickerBirthDate.Value, Double.Parse(tBSalary.Text), new Department(tBDpName.Text));
```
Comment (rodi0878) [inline INPTP_AppForFixing/EmployeeDialog.cs]: newB???

File: INPTP_AppForFixing/EmployeeDialog.cs [hunk 3]
```diff
@@ -1,59 +1,78 @@
...
+                Boss b = main.Bosses.FirstOrDefault(x => x.Id == Id);
```
Comment (rodi0878) [inline INPTP_AppForFixing/EmployeeDialog.cs]: b?

File: INPTP_AppForFixing/EmployeeDialog.cs [hunk 4]
```diff
@@ -1,59 +1,78 @@
...
+                }
+                else
+                {
+                    main.Bosses.RemoveWhere(x => x.Id == newB.Id);
+                    main.Bosses.Add(newB);
+                    b = newB;
+                }
                 main.OnBossesChange();
-                Close();
+                this.Close();
```
Comment (rodi0878) [inline INPTP_AppForFixing/EmployeeDialog.cs]: this is redundant

File: INPTP_AppForFixing/EmployeeDialog.cs [hunk 5]
```diff
@@ -1,59 +1,78 @@
...
-                                    MessageBoxButtons.OK, MessageBoxIcon.Error);
+            else
+            {
+                MessageBox.Show("Bad entry data inserted.", "Error",
+                                MessageBoxButtons.OK, MessageBoxIcon.Error);
             }
         }
 
-        private void EmployeeDialog_FormClosed(object sender, FormClosedEventArgs e)
+        private bool testOfValues()
```
Comment (rodi0878) [inline INPTP_AppForFixing/EmployeeDialog.cs]: test of what values and what are you testing? name of this method could be much better...

File: INPTP_AppForFixing/MainForm.cs [hunk 6]
```diff
@@ -11,10 +11,11 @@
 namespace INPTP_AppForFixing
 {
     public partial class MainForm : Form
-    {
-
+    {       
+        private HashSet<Boss> employee;
```
Comment (rodi0878) [inline INPTP_AppForFixing/MainForm.cs]: do we really need two different sets? I would expect that organizational structure is hierarchical and that i will see that structure in gui

File: INPTP_AppForFixing/MainForm.cs [hunk 7]
```diff
@@ -33,10 +34,10 @@ private void btnAddBoss_Click(object sender, EventArgs e)
             openAddDialog(false, "Create boss");            
         }
 
-        private void openAddDialog(bool emp, string text)
+        private void openAddDialog(bool emp, string text, Boss b = null)
```
Comment (rodi0878) [inline INPTP_AppForFixing/MainForm.cs]: b?

File: INPTP_AppForFixing/EmployeeDialog.cs (hunk 2 above)
Comment (jame581) [reply INPTP_AppForFixing/EmployeeDialog.cs]: Fixed - renamed to correct name

File: INPTP_AppForFixing/EmployeeDialog.cs (hunk 3 above)
Comment (jame581) [reply INPTP_AppForFixing/EmployeeDialog.cs]: Fixed - renamed to correct name

File: INPTP_AppForFixing/EmployeeDialog.cs (hunk 4 above)
Comment (jame581) [reply INPTP_AppForFixing/EmployeeDialog.cs]: Fixed

File: INPTP_AppForFixing/EmployeeDialog.cs (hunk 5 above)
Comment (jame581) [reply INPTP_AppForFixing/EmployeeDialog.cs]: Fixed - renamed to correct name (ValidFormInput)

File: INPTP_AppForFixing/MainForm.cs (hunk 6 above)
Comment (jame581) [reply INPTP_AppForFixing/MainForm.cs]: No we do not need two sets. I forget on hashSet in Boss.

File: INPTP_AppForFixing/MainForm.cs (hunk 7 above)
Comment (jame581) [reply INPTP_AppForFixing/MainForm.cs]: Fixed - renamed to correct name

**Comment by jame581** [comment]
I created patch for fixing your requirements. I added logic for add employee to boss like a bonus to patch

File: INPTP_AppForFixing/EmployeeDialog.cs [hunk 8]
```diff
@@ -1,59 +1,92 @@
...
 using System.Windows.Forms;
 
 namespace INPTP_AppForFixing
 {
     public partial class EmployeeDialog : Form
     {
         private MainForm main;
-        private bool employee=false;
+        private bool employee = false;
+        private Boss bossForEmpl;
```
Comment (rodi0878) [inline INPTP_AppForFixing/EmployeeDialog.cs]: Again name with abbreviation.

File: INPTP_AppForFixing/EmployeeDialog.cs [hunk 9]
```diff
@@ -1,59 +1,92 @@
...
         private void btnOk_Click(object sender, EventArgs e)
         {
-            if (testOfValues())
+            if (validFormInput())
             {
-                main.Bosses.Add(new Boss(Int32.Parse(tBID.Text), tBFirstName.Text, tBLastName.Text, tBJob.Text, dateTimePickerBirthDate.Value, Double.Parse(tBSalary.Text), new Department(tBDpName.Text)));
-                main.OnBossesChange();
+                if (employee)
+                {
+                    Employee temp = new Employee(int.Parse(tBID.Text), tBFirstName.Text, tBLastName.Text, tBJob.Text, dateTimePickerBirthDate.Value, Double.Parse(tBSalary.Text));
```
Comment (rodi0878) [inline INPTP_AppForFixing/EmployeeDialog.cs]: Please no temp, tmp, x, var, variables or variables with nonsense names.

File: INPTP_AppForFixing/EmployeeDialog.cs [hunk 10]
```diff
@@ -1,59 +1,92 @@
...
+                    bossForEmpl.InsertEmpl(temp);
+                    main.OnEmplChange();
+                }
+                else
+                {
+                    Boss newBoss = new Boss(int.Parse(tBID.Text), tBFirstName.Text, tBLastName.Text, tBJob.Text, dateTimePickerBirthDate.Value, Double.Parse(tBSalary.Text), new Department(tBDpName.Text));
+
+                    // Find boss in hashset by boss ID
```
Comment (rodi0878) [inline INPTP_AppForFixing/EmployeeDialog.cs]: comments that tells us what code does are bad comments. Comments should explain why you did this...

File: INPTP_AppForFixing/EmployeeDialog.cs [hunk 11]
```diff
@@ -1,59 +1,92 @@
...
-                                    MessageBoxButtons.OK, MessageBoxIcon.Error);
+            else
+            {
+                MessageBox.Show("Bad entry data inserted.", "Error",
+                                MessageBoxButtons.OK, MessageBoxIcon.Error);
             }
         }
 
-        private void EmployeeDialog_FormClosed(object sender, FormClosedEventArgs e)
+        private bool validFormInput()
```
Comment (rodi0878) [inline INPTP_AppForFixing/EmployeeDialog.cs]: functions/methods should have verb in name "isValidFormInput","testIfFormInputIsValid",...

File: INPTP_AppForFixing/MainForm.cs [hunk 12]
```diff
@@ -48,6 +39,18 @@ private void openAddDialog(bool emp, string text)
             }
         }
 
+        public void OnEmplChange()
```
Comment (rodi0878) [inline INPTP_AppForFixing/MainForm.cs]: no abbreviations

File: INPTP_AppForFixing/MainForm.cs [hunk 13]
```diff
@@ -33,12 +25,11 @@ private void btnAddBoss_Click(object sender, EventArgs e)
...
         }
 
-        private void openAddDialog(bool emp, string text)
+        private void openAddDialog(bool emp, string text, Boss bossToEdit = null)
         {
-            EmployeeDialog createWin = new EmployeeDialog(this, emp);
-            createWin.Text = "Create employee";
-            createWin.Show();
-            Enabled = false;
+            EmployeeDialog createWin = new EmployeeDialog(this, emp, bossToEdit);
```
Comment (rodi0878) [inline INPTP_AppForFixing/MainForm.cs]: no abbreviations

File: INPTP_AppForFixing/EmployeeDialog.cs (hunk 8 above)
Comment (jame581) [reply INPTP_AppForFixing/EmployeeDialog.cs]: Fixed

File: INPTP_AppForFixing/EmployeeDialog.cs (hunk 9 above)
Comment (jame581) [reply INPTP_AppForFixing/EmployeeDialog.cs]: Ok renamed.

File: INPTP_AppForFixing/EmployeeDialog.cs (hunk 10 above)
Comment (jame581) [reply INPTP_AppForFixing/EmployeeDialog.cs]: I understand

File: INPTP_AppForFixing/EmployeeDialog.cs (hunk 11 above)
Comment (jame581) [reply INPTP_AppForFixing/EmployeeDialog.cs]: OK

File: INPTP_AppForFixing/MainForm.cs (hunk 12 above)
Comment (jame581) [reply INPTP_AppForFixing/MainForm.cs]: ok

**Pull request merged**